File: lib/models/exercise.dart
class Exercise {
  String name;
  int series;
  String reps;
  String pause;
  double weight;

  Exercise({
    required this.name,
    required this.series,
    required this.reps,
    required this.pause,
    this.weight = 0,
  });

  Map<String, dynamic> toMap() {
    return {
      'name': name,
      'series': series,
      'reps': reps,
      'pause': pause,
      'weight': weight,
    };
  }

  factory Exercise.fromMap(Map<String, dynamic> map) {
    return Exercise(
      name: map['name'],
      series: map['series'],
      reps: map['reps'],
      pause: map['pause'],
      weight: map['weight'].toDouble(),
    );
  }
}

File: app/models/exercise.ts
export class Exercise {
    constructor(
        public name: string,
        public series: number,
        public reps: string,
        public pause: string,
        public weight: number = 0
    ) {}

    toJSON() {
        return {
            name: this.name,
            series: this.series,
            reps: this.reps,
            pause: this.pause,
            weight: this.weight
        };
    }

    static fromJSON(json: any): Exercise {
        return new Exercise(
            json.name,
            json.series,
            json.reps,
            json.pause,
            json.weight
        );
    }
}